[OpenMP][Clang][Driver] Removes opaque-offload-linker

nicebert opened the pull request:
Part of ongoing effort to get covergence with upstream llvm. Removes opaque-offload-linker from clang driver which was introduced after switching over to the new driver to enhance debugging.

nicebert committed:
[OpenMP][Clang][Driver] Removes opaque-offload-linker

Part of ongoing effort to get covergence with upstream llvm.
Removes opaque-offload-linker from clang driver which was introduced after switching over to the new driver to enhance debugging.

z1-cciauto commented:
PSDB Link: https://compiler-ci.amd.com/job/compiler-psdb-amd-staging/2681

ronlieb commented:
looks like it is getting close to landing:
one lit test to investigate/fix/update ...

[2025-11-06T09:50:19.297Z] Failed Tests (1):

[2025-11-06T09:50:19.297Z]   Clang :: Driver/openmp-offload-infer.c

CatherineMoore reviewed (commented):
The patch looks good (fairly straightforward). Approval pending passing tests.

nicebert committed:
[OpenMP] Fixes failing lit test

Removes check for removed flag -no-opaque-offload-linker after removing the opaque linker.

z1-cciauto commented:
PSDB Link: https://compiler-ci.amd.com/job/compiler-psdb-amd-staging/2750
